src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,62 @@
+package duke;
+
+import duke.command.Command;
+
+/**
+ * Encapsulates chatbot 'Duke'. Contains main method for chatbot
+ */
+
+public class Duke {
+    /** Unit that manages saved data of task list. */
+    private Storage storage;
+    /** List of tasks tracked by Duke. */
+    private TaskList tasks;
+    /** Unit that manages user interface of Duke. */
+    private Ui ui;
+
+    /**
+     * Initializes a Duke object.
+     *
+     * @param filePath Pathname of the file used for storing data.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the chatbot.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```

Comment:
Hi, overall ur Duke.java code looks really clean and organized. However, do you think putting the flag isExit in the Command class will make your code even cleaner. So, your while condition can be just !c.isExit(), and in the Command class if the command is actually "bye" u can call c.exit() which will mutate the isExit instance variable of the Command instance object to true.

Reply:
That make sense, I will see if I can integrate it after implementing JavaFX (assuming the code structure is still similar).
Thank you for the feedback!